src/convertToObject.js

```diff
@@ -12,7 +12,15 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString.split(';').reduce((acc, cur) => {
+    const [key, value] = cur.split(':');
+
+    if (/\S/.test(key) && /\S/.test(value)) {
+      acc[key.trim()] = value.trim();
+    }
```

Comment:
i meaaaan it works. But i was talking about trimming the key and value before the if block